data/logs_model/logs_producer/splunk_logs_producer.py

```diff
@@ -22,11 +23,33 @@ def __init__(
         url_scheme="https",
         verify_ssl=True,
         index_prefix=None,
+        ssl_ca_path=None,
     ):
+        connect_args = {
+            "host": host,
+            "port": port,
+            "token": bearer_token,
+            "scheme": url_scheme,
+            "verify": verify_ssl,
+        }
+        if verify_ssl:
```

Comment:
I think you need to create the context if `ssl_ca_path` is set (to something other than `""`), independent of `verify_ssl`.
Users who will configure a custom CA will usually want SSL verification, but in theory a user could set a custom CA and still skip SSL verification. In other words, the SSL context creation should happen in connection to the user telling Quay to use a custom CA when talking to Splunk.

Furthermore, I think you want to investigate the meaning of [`context.verify_mode`](https://docs.python.org/3/library/ssl.html#ssl.SSLContext.verify_mode). From what I gather, `verify_ssl = False` is equivalent to `context.verify_mode = CERT_NONE`, and `verify_ssl = True` is equivalent to `context.verify_mode = ssl.CERT_REQUIRED`.